enumerator/usb_netbsd.go

```diff
@@ -0,0 +1,40 @@
+//
+// Copyright 2014-2023 Cristian Maglie. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+//
+
+package enumerator
+
+import (
+	"strings"
+
+	"go.bug.st/serial"
+)
+
+func nativeGetDetailedPortsList() ([]*PortDetails, error) {
+	// Retrieve the port list
+	ports, err := serial.GetPortsList()
+	if err != nil {
+		return nil, &PortEnumerationError{causedBy: err}
+	}
+
+	var res []*PortDetails
+	for _, port := range ports {
+		details, err := nativeGetPortDetails(port)
+		if err != nil {
+			return nil, &PortEnumerationError{causedBy: err}
+		}
+		res = append(res, details)
+	}
+	return res, nil
+}
+
+func nativeGetPortDetails(portPath string) (*PortDetails, error) {
+	result := &PortDetails{Name: portPath}
+
+	if strings.Contains(result.Name, "U") {
+		result.IsUSB = true
+	}
+	return result, nil
+}
```

Comment:
It would be great to get the USB VID, PID, and `ISerial` number is this possible?

Reply:
> It would be great to get the USB VID, PID, and `ISerial` number is this possible?

As mentioned above.

Reply:
As mentioned above.

Reply:
As mentioned below.

Reply:
Indeed, they are just available precreated node. There is no sysfs on NetBSD yet, maybe have a better solution in the future.

Reply:
It is possible, what you need to do is to open `/dev/usb0` and issue the `DRVLISTDEV` ioctl. [example code in C](https://nxr.netbsd.org/xref/src/usr.sbin/usbdevs/usbdevs.c)
I can send a separate PR to add support for that after the basic support is merged.